### PR TITLE
🧪 Scout: add ICU parser edge case tests

### PR DESCRIPTION
**Pull request by cungminh2710**
Improved ICU parser test coverage by adding validation for MDX-style member component tags (containing dots and numbers) and negative plural offsets. These tests ensure that Hyperlocalise robustly handles modern frontend component patterns and valid but less common ICU message syntax.

---
*PR created automatically by Jules for task [16158431175174750477](https://jules.google.com/task/16158431175174750477) started by @cungminh2710*

### CODE REVIEW

File: internal/i18n/icuparser/parse_test.go [hunk 1]
```diff
@@ -413,6 +413,27 @@ func TestParseTypedFormatterShouldParseSkeletons(t *testing.T) {
...
+	tag, ok := elems[1].(TagElement)
+	if !ok {
+		t.Fatalf("expected tag element, got %T", elems[1])
+	}
+	if tag.Value != "My.Component.v1" {
+		t.Fatalf("unexpected tag value: %q", tag.Value)
+	}
+	if len(tag.Children) != 1 {
+		t.Fatalf("expected 1 child for My.Component.v1, got %d", len(tag.Children))
+	}
```
[the remark this answers is not in the document]
Comment (cungminh2710) [reply internal/i18n/icuparser/parse_test.go:437]: @jules fix
